src/Microsoft.DotNet.GenAPI/Shared/Filters/IntersectionFilter.cs

```diff
@@ -0,0 +1,67 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Collections.Generic;
+using Microsoft.CodeAnalysis;
+
+namespace Microsoft.DotNet.GenAPI.Shared;
+
+
+using IFilter = IAssemblySymbolFilter;
+
+public class IntersectionFilter : IFilter
+{
+    private readonly List<IFilter> _innerFilters = new List<IFilter>();
+
+    /// <inheritdoc />
+    public bool Include(INamespaceSymbol ns)
+    {
+        foreach (var innerFilter in _innerFilters)
+        {
+            if (!innerFilter.Include(ns)) return false;
+        }
+        return true;
+    }
+
+    /// <inheritdoc />
+    public bool Include(AttributeData at)
+    {
+        foreach (var innerFilter in _innerFilters)
+        {
+            if (!innerFilter.Include(at)) return false;
+        }
+        return true;
```

Comment:
You could probably us `Any()` or `All()`?

```suggestion
        return _innerFilters.All(f => f.Include(at));
```

Reply:
Same for the rest